<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.5">
  <compounddef id="classgmx_1_1test_1_1TestException" kind="class" prot="public">
    <compoundname>gmx::test::TestException</compoundname>
    <basecompoundref refid="classgmx_1_1GromacsException" prot="public" virt="non-virtual">gmx::GromacsException</basecompoundref>
    <includes local="no">testutils/testexceptions.h</includes>
      <sectiondef kind="public-func">
      <memberdef kind="function" id="classgmx_1_1test_1_1TestException_1aa3c79e2b74a27e68018a7e990d7bc7fc" prot="public" static="no" const="no" explicit="yes" inline="yes" virt="non-virtual">
        <type></type>
        <definition>gmx::test::TestException::TestException</definition>
        <argsstring>(const std::string &amp;reason)</argsstring>
        <name>TestException</name>
        <param>
          <type>const std::string &amp;</type>
          <declname>reason</declname>
        </param>
        <briefdescription>
<para>Creates a test exception object with the provided detailed reason. </para>        </briefdescription>
        <detaileddescription>
<para><parameterlist kind="param"><parameteritem>
<parameternamelist>
<parametername direction="in">reason</parametername>
</parameternamelist>
<parameterdescription>
<para>Detailed reason for the exception. </para></parameterdescription>
</parameteritem>
</parameterlist>
</para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" line="71" column="1" bodyfile="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" bodystart="71" bodyend="72"/>
      </memberdef>
      <memberdef kind="function" id="classgmx_1_1test_1_1TestException_1a3616ce31284d79ab68e8faa65330d5ec" prot="public" static="no" const="no" explicit="yes" inline="yes" virt="non-virtual">
        <type></type>
        <definition>gmx::test::TestException::TestException</definition>
        <argsstring>(const GromacsException &amp;base)</argsstring>
        <name>TestException</name>
        <param>
          <type>const <ref refid="classgmx_1_1GromacsException" kindref="compound">GromacsException</ref> &amp;</type>
          <declname>base</declname>
        </param>
        <briefdescription>
<para>Creates a test exception based on another <ref refid="classgmx_1_1GromacsException" kindref="compound">GromacsException</ref> object. </para>        </briefdescription>
        <detaileddescription>
<para><parameterlist kind="param"><parameteritem>
<parameternamelist>
<parametername direction="in">base</parametername>
</parameternamelist>
<parameterdescription>
<para>Exception to wrap.</para></parameterdescription>
</parameteritem>
</parameterlist>
<simplesect kind="see"><para><ref refid="testexceptions_8h_1abcf749f0c519a9f9d00a657593fc8260" kindref="member">GMX_THROW_WRAPPER_TESTEXCEPTION</ref> </para></simplesect>
</para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" line="80" column="1" bodyfile="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" bodystart="80" bodyend="81"/>
      </memberdef>
      <memberdef kind="function" id="classgmx_1_1test_1_1TestException_1a0eea517643f844209a6d0068321a9cf6" prot="public" static="no" const="yes" explicit="no" inline="yes" virt="virtual">
        <type>int</type>
        <definition>virtual int gmx::test::TestException::errorCode</definition>
        <argsstring>() const </argsstring>
        <name>errorCode</name>
        <reimplements refid="classgmx_1_1GromacsException_1a3d1cac3238594bbd9a483ccd7d57860d">errorCode</reimplements>
        <briefdescription>
<para>Returns the error code corresponding to the exception type. </para>        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" line="83" column="1" bodyfile="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" bodystart="83" bodyend="83"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
<para>Exception class for reporting errors in tests. </para>    </briefdescription>
    <detaileddescription>
<para> <libinternal /> This exception should be used for error conditions that are internal to the test, i.e., do not indicate errors in the tested code. </para>    </detaileddescription>
    <inheritancegraph>
      <node id="3295">
        <label>gmx::GromacsException</label>
        <link refid="classgmx_1_1GromacsException"/>
        <childnode refid="3296" relation="public-inheritance">
        </childnode>
      </node>
      <node id="3296">
        <label>std::exception</label>
      </node>
      <node id="3294">
        <label>gmx::test::TestException</label>
        <link refid="classgmx_1_1test_1_1TestException"/>
        <childnode refid="3295" relation="public-inheritance">
        </childnode>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="3298">
        <label>gmx::GromacsException</label>
        <link refid="classgmx_1_1GromacsException"/>
        <childnode refid="3299" relation="public-inheritance">
        </childnode>
      </node>
      <node id="3299">
        <label>std::exception</label>
      </node>
      <node id="3297">
        <label>gmx::test::TestException</label>
        <link refid="classgmx_1_1test_1_1TestException"/>
        <childnode refid="3298" relation="public-inheritance">
        </childnode>
      </node>
    </collaborationgraph>
    <location file="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" line="64" column="1" bodyfile="/home/jenkins/workspace/Release_workflow_master/gromacs-2018.3/src/testutils/testexceptions.h" bodystart="63" bodyend="84"/>
    <listofallmembers>
      <member refid="classgmx_1_1test_1_1TestException_1a0eea517643f844209a6d0068321a9cf6" prot="public" virt="virtual"><scope>gmx::test::TestException</scope><name>errorCode</name></member>
      <member refid="classgmx_1_1GromacsException_1a1c3f6e30d9dd5682013158d83b53a850" prot="public" virt="non-virtual"><scope>gmx::test::TestException</scope><name>getInfo</name></member>
      <member refid="classgmx_1_1GromacsException_1a1a17194de13545be2ae61aa7d95479e6" prot="protected" virt="non-virtual"><scope>gmx::test::TestException</scope><name>GromacsException</name></member>
      <member refid="classgmx_1_1GromacsException_1a1bd116912e67c096ae949ddae85afe06" prot="public" virt="non-virtual"><scope>gmx::test::TestException</scope><name>prependContext</name></member>
      <member refid="classgmx_1_1GromacsException_1adadfedffd28181f9fda74bf40e846b5c" prot="public" virt="non-virtual"><scope>gmx::test::TestException</scope><name>setInfo</name></member>
      <member refid="classgmx_1_1test_1_1TestException_1aa3c79e2b74a27e68018a7e990d7bc7fc" prot="public" virt="non-virtual"><scope>gmx::test::TestException</scope><name>TestException</name></member>
      <member refid="classgmx_1_1test_1_1TestException_1a3616ce31284d79ab68e8faa65330d5ec" prot="public" virt="non-virtual"><scope>gmx::test::TestException</scope><name>TestException</name></member>
      <member refid="classgmx_1_1GromacsException_1a75c448ba14a96a69f9a184b7e565ead9" prot="public" virt="virtual"><scope>gmx::test::TestException</scope><name>what</name></member>
      <member refid="classgmx_1_1GromacsException_1a8f78317f29da9cb560f76e79abc174b6" prot="public" virt="virtual"><scope>gmx::test::TestException</scope><name>~GromacsException</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
